server/manager.go

```diff
@@ -207,7 +207,7 @@ func (m *Manager) InitServer(data remote.ServerConfigurationResponse) (*Server,
 		Limits:      s.cfg.Build,
 	}
 
-	envCfg := environment.NewConfiguration(settings, s.GetEnvironmentVariables())
+	envCfg := environment.NewConfiguration(settings, s.GetEnvironmentVariables(), s.Config().Labels)
```

Comment:
Use the existing `Settings` struct to pass labels.  I'm not exactly sure why environment variables were separated in the first place.